tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -2383,6 +2383,9 @@ void CGMSHLSLRuntime::MarkRetTemp(CodeGenFunction &CGF, Value *V,
 void CGMSHLSLRuntime::MarkCallArgumentTemp(CodeGenFunction &CGF, llvm::Value *V,
                                            clang::QualType QualTy) {
   // Save object properties for call arg temp.
+  // Igonre V already in property map.
+  if (objectProperties.GetResource(V).isValid())
+    return;
```

Comment:
I understand how this can occur. It is possible that the QualTy match, but because of how AddValToPropertyMap is done, the assert will fire anyway. I think it's also possible that they don't match and the assert will rightly fire. In that case, would skipping the update to property map entirely miss some of the point of #3458? Might we want to merge the types instead? Or do we know for sure that they will always be the same?